system-addon/lib/TelemetryFeed.jsm

```diff
@@ -256,8 +353,13 @@ this.TelemetryFeed = class TelemetryFeed {
       case at.SAVE_SESSION_PERF_DATA:
         this.saveSessionPerfData(au.getPortIdOfSender(action), action.data);
         break;
+      case at.SYSTEM_TICK:
+        if (Date.now() - this._impressionStatsLastReset >= IMPRESSION_STATS_RESET_TIME) {
+          this.resetImpressionStats();
+        }
```

Comment:
Yes, it is intentional. However, after the second thought, I think initializing it to `Date.now()` would be preferred, since this timestamp isn't persisted in the prefs. Rebooting Firefox frequently (e.g. update restart) may introduce some undesired behaviors. Will change it to "Date.now()". Thanks!